control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go

```diff
@@ -955,6 +955,84 @@ func TestReconcileClusterVersion(t *testing.T) {
 	g.Expect(clusterVersion.Spec.Channel).To(BeEmpty())
 }
 
+func TestReconcileClusterVersionWithDisabledCapabilities(t *testing.T) {
+	hcp := &hyperv1.HostedControlPlane{
+		Spec: hyperv1.HostedControlPlaneSpec{
+			ClusterID: "test-cluster-id",
+			Capabilities: &hyperv1.Capabilities{
+				DisabledCapabilities: []hyperv1.OptionalCapability{
+					hyperv1.ImageRegistryCapability,
+				},
+			},
+		},
+	}
+	testOverrides := []configv1.ComponentOverride{
+		{
+			Kind:      "Pod",
+			Group:     "",
+			Name:      "test",
+			Namespace: "default",
+			Unmanaged: true,
+		},
+	}
+	clusterVersion := &configv1.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "version",
+		},
+		Spec: configv1.ClusterVersionSpec{
+			ClusterID: "some-other-id",
+			Capabilities: &configv1.ClusterVersionCapabilitiesSpec{
+				AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{
+					"foo",
+					"bar",
+				},
+			},
+			Channel: "fast",
+			DesiredUpdate: &configv1.Update{
+				Version: "4.12.5",
+				Image:   "example.com/imagens/image:latest",
+				Force:   true,
+			},
+			Upstream:  configv1.URL("https://upstream.example.com"),
+			Overrides: testOverrides,
+		},
+	}
+	fakeClient := fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(clusterVersion).Build()
+	g := NewWithT(t)
+	r := &reconciler{
+		client:                 fakeClient,
+		CreateOrUpdateProvider: &simpleCreateOrUpdater{},
+	}
+	err := r.reconcileClusterVersion(context.Background(), hcp)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = fakeClient.Get(context.Background(), client.ObjectKeyFromObject(clusterVersion), clusterVersion)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	expectedCapabilities := &configv1.ClusterVersionCapabilitiesSpec{
+		BaselineCapabilitySet: configv1.ClusterVersionCapabilitySetNone,
+		AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{
+			configv1.ClusterVersionCapabilityBuild,
```

Comment:
given we use this in many tests, we might want to have a common variable with the full list, then for comparison we subtract ClusterVersionCapabilityImageRegistry from that common source?

Reply:
Now that I removed the e2e tests from this PR, this is the only place that builds this list, so I'll defer this to the follow-up PR.